### ISSUE TITLE
About  esrgan Parameter

### ISSUE THREAD

**Issue by CrystalWong2** [xinntao/BasicSR#40]
I saw some parameter in train_esrgan.json，but I am not sure the meaning. So pls can you give some detail to help me?
 
1.Like beta1_G,beta1_D,lr_gamma in train_esrgan.json?

2.I saw in the code have the training model for wgan-gp,so can it work? Because I saw ReadMe just delete this part. So I am not sure whether you change the code or not? Did you do the experiment to use the D as wgan-gp? If you did it before, could you tell me the perceptual quality use the D as wgan-gp?

3.I am a new in Neural Networks, and never try a model in image. So I want to know at least how many image can I build the esrgan model if I use 23 DDRB? And if I don't change the architecture, can I use my pre-train mode as pretrain_model_G,then change the parameter and run the code again to get a better network?

Please help，thank you so much...

**Comment by xinntao** [comment]
Hi @CrystalWong2 
1. A general explanation of options could be found in [options-README](https://github.com/xinntao/BasicSR/blob/master/codes/options/README.md).
beta1_G, geta1_D are the options of Adam optimizer. In many GAN papers (like DCGAN), they usually set them to 0.5. And the default value is 0.9.
lr_gamma is the option for MultiStepLR learning scheme. Please refer to pytorch doc.

2. The codes of WGAN-GP is ready for running. I tried with WGAN-GP, but there is no obvious improvements and the training time is longer than vanilla GAN or Relativistic GAN. Therefore, I do not use them but just keep the codes in this repo.

3. DIV2K is OK. Training a model with 23 RRDB is too slow for me. You may start from resnet structure with 16 RBs. You can also use your own structures and pre-trained models.

**Comment by CrystalWong2** [comment]
@xinntao Thanks for your reply..I have a small question. That is according to question 2 above，so can I change the gan-type in esrgan.json? Like you told me that  vanilla GAN or Relativistic GAN. But I can not find the code to explain vanilla or Relativistic GAN. Because I want to try the Relativistic GAN instead.

**Comment by CrystalWong2** [comment]
@xinntao I saw some note in architecture.json # VGG style Discriminator with input size 128*128. Is this a mandatory requirement? Will it affect the network if I change the input size like 118*118 ?

**Comment by CrystalWong2** [comment]
@xinntao Thanks for your help！
